lib/iris/_concatenate.py

```diff
@@ -1534,9 +1556,22 @@ def _build_data(self):
 
         """
         skeletons = self._skeletons
-        data = [skeleton.data for skeleton in skeletons]
 
-        data = concatenate_arrays(data, self.axis)
+        if all(skeleton.data is None for skeleton in skeletons):
+            data = None
+        else:
+            data = []
+            for skeleton in skeletons:
+                if skeleton.data is None:
+                    skeleton_data = da.ma.masked_array(
+                        data=da.zeros(skeleton.shape),
```

Comment:
The default datatype for a dask array is `float64`

I tested concatenating a dataless cube with a dataful one (that had a data dtype of `float32`) and the resultant concatenated cube had a data dtype of `float64` - i.e. **it had artificially elevated the original datatype**. This would be an _unexpected side effect_ of concatenation with datalless cubes?

This is due to numpy honouring the highest level of precision when concatenating arrays.
I testing changing this to:

```python
       data=da.zeros(skeleton.shape, dtype=np.float16),
```
and this solved the issue.

However, it might be best to set this to a small integer so that we don't accidentally turn integer data arrays in to floats (ints will automatically get elevated to floats when concatenated with another float array), i.e.

```suggestion python
                        data=da.zeros(skeleton.shape, dtype=np.int8)
```

The other alternative is to check the datatype of the input arrays of course, but if they are mixed types then you have to figure out what type to use for you mask!